tests/test_variant.h

```diff
@@ -105,6 +105,600 @@ TEST_CASE("[Variant] Writer and parser float") {
 
 	CHECK_MESSAGE(b64_float_parsed == 340282001837565597733306976381245063168.0, "Should not overflow.");
 }
+
+TEST_CASE("[Variant] To Bool from Int,Float,String,Vec2,Vec2i,Vec3,Vec3i and Color") {
+	Variant int_v = 0;
+	Variant bool_v = true;
+	int_v = bool_v; // int_v is now a bool
+	CHECK(int_v == Variant(true));
+	bool_v = false;
+	int_v = bool_v;
+	CHECK(int_v == Variant(false));
+
+	Variant float_v = 0.0f;
+	bool_v = true;
+	float_v = bool_v;
+	CHECK(float_v == Variant(true));
+	bool_v = false;
+	float_v = bool_v;
+	CHECK(float_v == Variant(false));
+
+	Variant string_v = "";
+	bool_v = true;
+	string_v = bool_v;
+	CHECK(string_v == Variant(true));
+	bool_v = false;
+	string_v = bool_v;
+	CHECK(string_v == Variant(false));
+
+	Variant vec2_v = Vector2(0, 0);
+	bool_v = true;
+	vec2_v = bool_v;
+	CHECK(vec2_v == Variant(true));
+	bool_v = false;
+	vec2_v = bool_v;
+	CHECK(vec2_v == Variant(false));
+
+	Variant vec2i_v = Vector2i(0, 0);
+	bool_v = true;
+	vec2i_v = bool_v;
+	CHECK(vec2i_v == Variant(true));
+	bool_v = false;
+	vec2i_v = bool_v;
+	CHECK(vec2i_v == Variant(false));
+
+	Variant vec3_v = Vector3(0, 0, 0);
+	bool_v = true;
+	vec3_v = bool_v;
+	CHECK(vec3_v == Variant(true));
+	bool_v = false;
+	vec3_v = bool_v;
+	CHECK(vec3_v == Variant(false));
+
+	Variant vec3i_v = Vector3i(0, 0, 0);
+	bool_v = true;
+	vec3i_v = bool_v;
+	CHECK(vec3i_v == Variant(true));
+	bool_v = false;
+	vec3i_v = bool_v;
+	CHECK(vec3i_v == Variant(false));
+
+	Variant col_v = Color(0.5f, 0.2f, 0.75f);
+	bool_v = true;
+	col_v = bool_v;
+	CHECK(col_v == Variant(true));
+	bool_v = false;
+	col_v = bool_v;
+	CHECK(col_v == Variant(false));
+}
+
+TEST_CASE("[Variant] To Int from Bool,Float,String,Vec2,Vec2i,Vec3,Vec3i and Color") {
+	Variant bool_v = false;
+	Variant int_v = 2;
+	bool_v = int_v; // Now bool_v is int
```

Comment:
You can check this explicitly with:
```suggestion
	bool_v = int_v;
	CHECK(bool_v.get_type() == Variant::TYPE_BOOL);
```

Reply:
I have added the explicit checking in place of the second Check Statement in each conversion. I have also kept the initial tests in the first Check Statement of each conversion just to check that the values are as expected.